lib/rules/sort-vars.js

```diff
@@ -17,21 +17,23 @@ module.exports = function(context) {
     return {
         "VariableDeclaration": function(node) {
             node.declarations.reduce(function(memo, decl) {
-                if (decl.id.type !== "ObjectPattern" && decl.id.type !== "ArrayPattern") {
-                    var lastVariableName = memo.id.name,
-                        currenVariableName = decl.id.name;
-
-                    if (ignoreCase) {
-                        lastVariableName = lastVariableName.toLowerCase();
-                        currenVariableName = currenVariableName.toLowerCase();
-                    }
-
-                    if (currenVariableName < lastVariableName) {
-                        context.report(decl, "Variables within the same declaration block should be sorted alphabetically");
-                        return memo;
-                    } else {
-                        return decl;
-                    }
+                if (decl.id.type === "ObjectPattern" || decl.id.type === "ArrayPattern") {
```

Comment:
Alternatively I could have used

```js
node.declarations.filter(function (decl) { var type = decl.id.type; return type !== "ObjectPattern" && type !== "ArrayPattern"; }).reduce
```

But this version (the one without `filter`) seems a bit more efficient. What do you think?